src/main/java/tech/stackable/hadoop/StackableGroupMapper.java

```diff
@@ -0,0 +1,142 @@
+package tech.stackable.hadoop;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.GroupMappingServiceProvider;
+import org.apache.hadoop.util.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+
+public class StackableGroupMapper implements GroupMappingServiceProvider {
+    public static final String OPA_MAPPING_URL_PROP = "hadoop.security.group.mapping.opa.url";
+    public static final String OPA_MAPPING_GROUP_NAME_PROP = "hadoop.security.group.mapping.opa.list.name";
+    // response base field: see https://www.openpolicyagent.org/docs/latest/rest-api/#response-message
+    public static final String OPA_RESULT_FIELD = "result";
+    public final String mappingGroupName;
+    private final Logger LOG = LoggerFactory.getLogger(StackableGroupMapper.class);
+    private final HttpClient httpClient = HttpClient.newHttpClient();
+    private final ObjectMapper json;
+    private URI opaUri = null;
+
+    public StackableGroupMapper() {
+        Configuration configuration = new Configuration();
```

Comment:
This is a somewhat expensive operation. Can you please make sure (e.g. by producing logs) that this is only called once?
Even better would be some sort of singleton or whatnot, as I will need it anyways in a authorizer function that is called frequently

Reply:
Implemented a single-element enum (which guarantees only being called once). 61ca906.

Reply:
Awesome thanks!